### PR TITLE
Support posix shells in uninstall script

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Refactor the `uninstall-thin-edge_io.sh` script to use `#!/bin/sh` (a posix compatible script), to make it more portable in different shells (not just bash). This is applicable for other linux distributions such as Alpine Linux.

It also provides a more consistent user experience as the `get-thin-edge_io.sh` already uses the `#!/bin/sh` shebang, so the uninstall can be run the same way the install script is run.

Included an example on how to run the uninstall as a one-liner as well.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

For references, the main changes when switching from bash to a posix shell format:
* Remove usage of bash arrays (as this is not posix, so not supported in `/bin/sh`)

### CODE REVIEW

**Review by PradeepKiruvale — APPROVED** [review]
LGTM, but didn't we remove the `c8y-remote-access` plugin from the `install` script? Is it ok to have it here?

**Comment by reubenmiller** [comment]
> LGTM, but didn't we remove the `c8y-remote-access` plugin from the `install` script? Is it ok to have it here?

Yes it is ok here because it is wrapped in the `dpkg -s <package> 2>/dev/null` call.

File: docs/src/howto-guides/002_installation.md [hunk 1]
```diff
@@ -164,6 +164,12 @@ Use uninstall script as shown below just to `remove` the `thin-edge.io` packages
 ```
 > Note: Removes just the thin-edge.io packages and does not remove the `configuration` files.
 
+The same thing can also be executed using a one-liner to download and run the script.
```
Comment (albinsuresh) [inline docs/src/howto-guides/002_installation.md:167]: I'd keep this one-liner command at the beginning of this uninstallation section itself. Let's keep the easiest thing first for the user. That's what we have done for the installation as well.

File: docs/src/howto-guides/002_installation.md [hunk 2]
```diff
@@ -172,6 +178,12 @@ associated with these thin-edge.io packages.
 ./uninstall-thin-edge_io.sh purge
 ```
 
+The same thing can also be executed using a one-liner to download and run the script.
```
Comment (albinsuresh) [inline docs/src/howto-guides/002_installation.md:181]: Same comment as above.

File: docs/src/howto-guides/002_installation.md (hunk 1 above)
Comment (reubenmiller) [reply docs/src/howto-guides/002_installation.md:169]: The problem is that having it is not easy having the one-liner at the beginning as we still need two different instances of it (one for removing and one for purging). 

If we set the syntax, then the code snippet is not convenient for a copy/paste as they need to edit it before executing it.

```sh
curl -sSL https://raw.githubusercontent.com/thin-edge/thin-edge.io/main/uninstall-thin-edge_io.sh | sudo sh -s <remove|purge>
```

So I just added a note to the user that there are one-liners available in the beginning section. Commit ac66afa8a515a82bc39f4d0212eb38d027ddefc3

Comment (reubenmiller) [reply docs/src/howto-guides/002_installation.md:169]: So basically the uninstall script is a bit different to the install script, as the uninstall script has a required positional argument.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**